src/Validator/Constraints/PhoneNumber.php

```diff
@@ -36,10 +36,12 @@ class PhoneNumber extends Constraint
 
     public const INVALID_PHONE_NUMBER_ERROR = 'ca23f4ca-38f4-4325-9bcc-eb570a4abe7f';
 
-    protected static $errorNames = [
+    protected const ERROR_NAMES = [
         self::INVALID_PHONE_NUMBER_ERROR => 'INVALID_PHONE_NUMBER_ERROR',
     ];
 
+    protected static $errorNames = self::ERROR_NAMES;
```

Comment:
IMHO this can be removed:

If people extend this class, changing its value won't have any effect (because `ERROR_NAMES` const has precedence over this static variable)

If you really want keeping BC, you should override the `getErrorName` method and take into account code that extends `ERROR_NAMES` and/or `$errorNames`. 
But IMHO that does not worth it, and you should remove simply remove that static variables

Reply:
I took a look deeper at what we're looking at here.

So [it's a deprecation of Symfony 6.1](https://github.com/symfony/symfony/blob/6.1/UPGRADE-6.1.md#validator) that comes with this new constant.

But removing the static property is not actually required and it will impact the validator for people using Symfony < 6.1. (which is today most people!)

So why remove this parameter? (I mean, using the same value is optional to me, but keeping it duplicated makes sense)

Reply:
Because the constant has precedence over the static property.
That means: if people extend `PhoneNumber` to change the value of `$errorNames` it won't work because Symfony will use `ERROR_NAMES` and ignore `$errorNames`, and that's also a BC break.

if you really want a BC solution, it should be something like (not tested and I'm not sure I've covered all the cases)

```php
    protected const ERROR_NAMES = [....];
    protected const $errorNames = static::ERROR_NAMES;

    public static function getErrorName(string $errorCode): string
    {
        // using self instead of static on purpose
        if (isset(self::ERROR_NAMES[$errorCode])) {
          // user unset the value using const 
          if (!isset(static::ERROR_NAMES[$errorCode])) {
            throw new InvalidArgumentException(sprintf('The error code "%s" does not exist for constraint of type "%s".', $errorCode, static::class));
          }

          // user unset the value using static var
          if (!isset(static::$errorNames[$errorCode])) {
            trigger_deprecation('symfony/validator', '6.1', 'The "%s::$errorNames" property is deprecated, use protected const ERROR_NAMES instead.', static::class);

            throw new InvalidArgumentException(sprintf('The error code "%s" does not exist for constraint of type "%s".', $errorCode, static::class));
          }

          // user changed the value using const
          if (static::ERROR_NAMES[$errorCode]) !== self::ERROR_NAMES[$errorCode])) {
            return static::ERROR_NAMES[$errorCode];
          }

          // user changed the value using static var
          if (static::$errorNames[$errorCode]) !== self::$errorNames[$errorCode])) {
            trigger_deprecation('symfony/validator', '6.1', 'The "%s::$errorNames" property is deprecated, use protected const ERROR_NAMES instead.', static::class);

            return static::$errorNames[$errorCode];
          }
        }

        return parent::getErrorName($errorCode);
    }
```

note that Symfony itself used :thinking:. Maybe that BC break is more acceptable than removing the property...

https://github.com/symfony/symfony/blob/d09634946b0e9e7035e138b52d6260b2bf7b783c/src/Symfony/Component/Form/Extension/Validator/Constraints/Form.php#L29-L32

Anyway, I've no strong opinion about this, I'm in the team composition over inheritance :wink:

Reply:
@norkunas sorry for making you go and back. But we're finally in an agreement!